common/src/traversal.rs

```diff
@@ -42,25 +42,25 @@ pub fn ensure_nearby(graph: &mut Graph, start: &Position, distance: f64) {
 pub fn nearby_nodes(
     graph: &Graph,
     start: &Position,
-    distance: f64,
+    distance: f32,
 ) -> Vec<(NodeId, na::Matrix4<f32>)> {
     struct PendingNode {
         id: NodeId,
-        transform: na::Matrix4<f64>,
+        transform: na::Matrix4<f32>,
     }
 
     let mut result = Vec::new();
-    let mut pending = Vec::<PendingNode>::new();
+    let mut pending = VecDeque::<PendingNode>::new();
```

Comment:
Is this change relevant?

Reply:
It is. When switching `traversal` to use `f32` instead of `f64`, I found out that it was using a depth-first search instead of a breadth-first search. This line of code is part of the change to switch it to a breadth-first search.

In this context, depth-first searches are more numerically unstable, since long chains of nodes cause numerical errors to propagate, and that became highly apparent when trying to use `f32` here (resulting in NaNs).

Reply:
If the specific strategy is important, consider explaining why with inline comments. In an ideal world this would also be a separate commit, preceding the changeover to f32.

Reply:
That's a fair point. I'll be an idealist here and split the commits.

Reply:
I'd like to see a little more detail in the comment as to *why* breadth-first is more stable. In a tree, this wouldn't be the case, because there's exactly one path from the root to each node anyway. I'm guessing the difference is that this graph contains cycles, so depth first can take a (much) longer than minimal path to a node?

Reply:
Got it. I'll flesh out the comment a bit. Your guess is correct.

Reply:
I've updated the wording to the following:
> We do a breadth-first instead of a depth-first traversal here to ensure that we take the minimal path to each node. This greatly helps prevent error from accumulating due to hundreds of transformations being composed.

This comment appears in two functions: `ensure_nearby` and `nearby_nodes`, so I've biased towards shorter comments. However, I'm wondering if I should just mention this as a more general principle at the top of the file.

Please let me know if you have any concerns with this wording and if I should flesh it out further.

Reply:
LGTM, thanks! Any sort of mention that a minimal path isn't always taken otherwise gets the point across, IMO.

Reply:
Awesome! I'll go ahead and turn on auto-merge (rebase).